src/irmin-pack/utils.ml

```diff
@@ -0,0 +1,80 @@
+type 'a fixed_width_fmt = 'a Fmt.t * int
+
+(** Pretty-printer for byte counts *)
+let pp_bytes : int64 fixed_width_fmt =
+  (* Round down to the nearest 0.1 *)
+  let trunc f = Float.trunc (f *. 10.) /. 10. in
+  let pp ppf i =
+    match Int64.to_float i with
+    | n when n < 1024. -> Fmt.pf ppf "%6.1f B  " (trunc n)
+    | n when n < 1024. ** 2. -> Fmt.pf ppf "%6.1f KiB" (trunc (n /. 1024.))
+    | n when n < 1024. ** 3. ->
+        Fmt.pf ppf "%6.1f MiB" (trunc (n /. (1024. ** 2.)))
+    | n -> Fmt.pf ppf "%6.1f GiB" (trunc (n /. (1024. ** 3.)))
+  in
+  (pp, 10)
+
+module Progress : sig
```

Comment:
that progress bar library looks really great, we should use it everywhere ;-)

Reply:
I decided I was tired of re-implementing this logic each time :wink: